src/pages/profile.tsx

```diff
@@ -37,21 +35,11 @@ interface Props {
 const nameSample = 'Mohammad Ali';
 const emailSample = 'mohammadali@quran.com';
 const ProfilePage: NextPage<Props> = ({ chaptersData }) => {
+  useRequireAuth();
   const dispatch = useDispatch();
   const { t, lang } = useTranslation();
   const router = useRouter();
-
-  const {
-    data: userData,
-    isValidating,
-    error,
-  } = useSWR(isLoggedIn() ? makeUserProfileUrl() : null, getUserProfile);
-
-  useEffect(() => {
-    if (!isLoggedIn()) {
-      router.replace('/login');
-    }
-  }, [router]);
+  const { user, isLoading, error } = useCurrentUser();
```

Comment:
1. this only fetches once, correct?
2. Don't we have an existing utility to get the user data? Might want to check with Osama